server/fishtest/views.py

```diff
@@ -1434,7 +1435,8 @@ def tests(request):
             ("Expires", "0"),
         )
     )
-    if int(request.params.get("page", 1)) > 1:
+    page_param = request.params.get("page", "")
+    if not page_param.isdigit() or int(page_param) > 1:
```

Comment:
```diff
diff --git a/server/fishtest/views.py b/server/fishtest/views.py
index cb3fd7e..d1443fd 100644
--- a/server/fishtest/views.py
+++ b/server/fishtest/views.py
@@ -1317,7 +1317,8 @@ def get_paginated_finished_runs(request):
     yellow_only = request.params.get("yellow_only", False)
     ltc_only = request.params.get("ltc_only", False)

-    page_idx = max(0, int(request.params.get("page", 1)) - 1)
+    page_param = request.params.get("page", "")
+    page_idx = max(0, int(page_param) - 1 if page_param.isdigit() else 0)
     page_size = 25

     finished_runs, num_finished_runs = request.rundb.get_finished_runs(
@@ -1374,7 +1375,8 @@ def tests_user(request):
     )
     username = request.matchdict.get("username", "")
     response = {**get_paginated_finished_runs(request), "username": username}
-    if int(request.params.get("page", 1)) == 1:
+    page_param = request.params.get("page", "")
+    if not page_param.isdigit() or int(page_param) <= 1:
         response["runs"] = request.rundb.aggregate_unfinished_runs(
             username=username,
         )[0]
@@ -1434,7 +1436,8 @@ def tests(request):
             ("Expires", "0"),
         )
     )
-    if int(request.params.get("page", 1)) > 1:
+    page_param = request.params.get("page", "")
+    if page_param.isdigit() and int(page_param) > 1:
         # page 2 and beyond only show finished test results
         return get_paginated_finished_runs(request)
```

Reply:
Are you sure about the last one
```
+    page_param = request.params.get("page", "")
+    if page_param.isdigit() and int(page_param) > 1:
```
this check for isdigit appears to be redundant, so no need for it right?

Reply:
The proposed code deals with "https://dfts-0.pigazzini.it/tests?page=qfdfasd" as well. I thought that that was your intention in using `isdigit()`

Reply:
BTW nn pagination needs the same changes.
"https://dfts-0.pigazzini.it/nns?page=sdcsdjbddve" crash

Reply:
Makes sense pushed.